### PR TITLE
SMASHED and SLAMMED Upgrade for the Summon Pitbulls spell.

### PR DESCRIPTION
**Pull request by Eneocho**
# Have you ever thought.. Hmm, those pitbulls I summon sure could be more smashed and slammed...
## Hi, I'm Arch-Wizard Derekus Baumicus, The Allmaker, and I'm declaring WAR on NT stations and non-wizards with the new SMASHED and SLAMMED Upgrade!
average pitbull wizard

**BANG BANG BANG! And 3 Toadline Pitbulls are summonned! Just make sure to UPGRADE the spell before using it.**
# AND YES! THIS IS EXACTLY HOW THEY LOOK!
## What do they do? Why should I upgrade?
Excellent question! A toadline pitbull is so SMASHED and SLAMMED that they can **_pass through AIRLOCKS_**! 
With some effort, it's **NOT INSTANTANEOUS**, but they sure can crawl through! They're so SMASHED that they can even walk through each other!
The boys at the wizard towerstation even managed to make them bite ever so slightly harder! They do move a bit slower though, what with their stumpy legs.
## But why?
It allows You, yes **YOU**!, to focus harder on the pitbulls! It does mean you have to spend more points on them meaning you have less for emergency escape spells, but we both know you were not planning on that when you bought the Toadlines!
## Are you SURE these spells work fine, sir? Last time we tried one of these we almost summoned Nar's-
FRET NOT! These spells were tested with High intensity sieges against NanoTrasen stations! Our boys in the lab robes even made sure that you do NOT summon the traditional pitbull when casting, and that upgrading works as intended!
:cl:
 * rscadd: Wizards can upgrade the Pitbull spell to summon Toadline Pitbulls for 20 spell points. Toadline Pitbulls move slightly slower than a regular pitbull, but have the advantage of being able to pass through airlocks and other mobs unimpeded (the airlock pass is NOT instantaneous, it takes a bit). They also do 1 more Brute damage on average compared to a regular Pitbull.

### CODE REVIEW

File: code/modules/spells/aoe_turf/conjure/pitbulls.dm [hunk 1]
```diff
@@ -18,12 +18,33 @@
...
+	. = "You have perfected the SMASHED and SLAMMED summon."
+
+/spell/aoe_turf/conjure/pitbull/invocation(mob/user, list/targets)
+	if(empowered)
+		invocation = pick("P'MPY S'N 'PP", "P'MPY S'N 'PP", "R'V'R'D' K'NN'LS", "BL'DSK'LL") 
+	..()
 
 var/list/pitbulls_exclude_kinlist = list() //all pitbulls go in here so pitbulls won't attack other pitbulls when feeling treacherous (and instead attack the wizard)
 
+/spell/aoe_turf/conjure/pitbull/perform(mob/user = usr, skipcharge = 0, list/target_override, var/ignore_timeless = FALSE, var/ignore_path = null)
```
Comment (Eneocho) [inline code/modules/spells/aoe_turf/conjure/pitbulls.dm:36]: not sure if this specific change will break stuff cause it didn't cause any issues as a plain perform() call on local, but it failed the linter check here without all that stuff in the (), but I can't test that today

**Comment by CrazyAmphibian** [comment]
\>1 more brute damage on average
why even?

**Comment by jwhitak** [comment]
image image



dog

**Comment by Eneocho** [comment]
>1 more brute damage on average why even?

it's a very small increase, but it stacks when you have more than 1 pitbull biting you simultaneously.
I CAN make them markedly stronger, but do you really want pitbulls that do 10 damage per bite? in practical terms i just upped the minimum and maximum damage values by 1, but that can be upped a bit more, say, by 3 instead of 1
I'm open to suggestions, cause while it is an almost negligible damage buff, it does stack up. Look at it this way, that 1 damage increase is a 25% damage increase PER pitbull (compared to the regular)

**Comment by Lumbermancer** [comment]
Sure why not.

**Comment by Lumbermancer** [comment]
Hats Center One 2024-06-26 000354
Sure why not. It's my go to spell.

File: code/modules/mob/living/simple_animal/hostile/pitbull.dm [hunk 2]
```diff
@@ -91,3 +91,41 @@
...
+	
+/mob/living/simple_animal/hostile/pitbull/smashednslammed/summoned_pitbull/death(var/gibbed = FALSE)
+	..()
+	if(!gibbed)
+		if(prob(90))
+			animate(src, alpha = 0, time = 4 SECONDS)
+			spawn(4 SECONDS)
+				qdel(src)
+		else
+			gib()
```
Comment (west3436) [inline code/modules/mob/living/simple_animal/hostile/pitbull.dm:131]: This is unnecessary as this behavior is already inherited from the parent proc.

File: code/modules/mob/living/simple_animal/hostile/pitbull.dm [hunk 3]
```diff
@@ -91,3 +91,41 @@
...
+		"Help I can't breathe",
+		"Bloodskull!",
+		"Two times Pimpy...",
+		"Three times Bape!",
+		"Not a family tree, but a family circle.")
+
+/mob/living/simple_animal/hostile/pitbull/smashednslammed/summoned_pitbull
+	faction = "wizard" // so they get along with other wizard mobs
+	meat_type = /obj/item/weapon/ectoplasm //a magical dog
+
```
Comment (west3436) [inline code/modules/mob/living/simple_animal/hostile/pitbull.dm:122]: Unnecessary as this is already inherited from the parent.

**Review by west3436 — CHANGES_REQUESTED** [review]
10/10 PR just remove the redundant code

File: code/modules/mob/living/simple_animal/hostile/pitbull.dm (hunk 2 above)
Comment (Eneocho) [reply code/modules/mob/living/simple_animal/hostile/pitbull.dm:131]: Smashed and slammed is not a child of the summoned pitbulls (which would make that code redundant), it's a child of regular pitbulls, meaning the summoned one is a child of the toadline pitbulls (which in turn is a child of regular pitbulls), it's not redundant code.
As for why have regular non summonable smashed pitbulls, there are some non wizard sources, like the syndie pokeballs or the goldcore reactions.

File: code/modules/mob/living/simple_animal/hostile/pitbull.dm (hunk 3 above)
Comment (Eneocho) [reply code/modules/mob/living/simple_animal/hostile/pitbull.dm:122]: Ditto

File: code/modules/mob/living/simple_animal/hostile/pitbull.dm (hunk 2 above)
Comment (west3436) [reply code/modules/mob/living/simple_animal/hostile/pitbull.dm:131]: Christ, that's a mess. I won't force you to, but cleaning up code here for consistency would be nice.

**Comment by SECBATON-GRIFFON** [comment]
upboated just for that derek bum image alone

**Comment by Kromkar** [comment]
3x Pimpy 
3x BAPE

**Comment by SonixApache** [comment]
remind me to add wizard lab robes at some point
sufficiently advanced technology, etc

also i see TG items in the OP image so you're getting a 👎

**Pull request merged**